src/client/components/equalizer/equalizer.js

```diff
@@ -0,0 +1,200 @@
+import React, {Component} from 'react';
+import Hark from 'hark';
+import PropTypes from 'prop-types';
+import Graphicequaliser from './canvasequalizer/canvasEqualizer';
+import PlayButton from './playbutton/playButton';
+import Uploadbutton from './upload/uploadButton';
+import Infoabouttrack from './upload/infoaboutfile/infoAboutUploadFile';
+import Streambutton from './streambutton/streamButton';
+import {connect} from 'react-redux';
+import {createAudioData, playPauseSoundFromFile, createStreamData, startMuteStreamAudio, mergeCanvasWidth} from '../../actions/audioActions';
+
+class Equalizer extends Component {
+  state = {
+    analyser: this.props.audioData.analyser,
+    ctx: undefined,
+    numPoints: undefined,
+    heightArray: undefined
+  }
+
+  componentDidMount() {
+    let canvasEl = document.querySelector("canvas").getContext("2d");
+    let numPoints = this.state.analyser.frequencyBinCount - 80;
+    let uint8Array = new Uint8Array(numPoints);
+    this.setState({
+      ctx: canvasEl,
+      numPoints: numPoints,
+      heightArray: uint8Array
+    });
+    this.detectStreamSoundFromMicrophon();
+  }
+
+  createSoundStream = (stream) => {
+    let {audioContext: context} = this.props.audioData;
+    let audioLineIn = new Audio();
+
+    audioLineIn.srcObject = stream;
+    audioLineIn.muted = true;
+    //hark - JS module that listens to an audio stream, and emits events indicating whether the user is speaking or not 
+    let options = {};
+    let speechEvents = Hark(stream, options);
+    let htmlinfo = document.getElementById("stream_detecting");
+    speechEvents.on('speaking', () => {
+      htmlinfo.innerHTML = `speaking`;
+    });
+    speechEvents.on('stopped_speaking', () => {
+      htmlinfo.innerHTML = `no stream detekting`;
+    });
+    let sourceStream = context.createMediaStreamSource(stream);
+    this.props.createStreamData({
+      audioLineIn: audioLineIn,
```

Comment:
in es6 you don't need to write key: value pairs if they are the same, please check this in all code you are committing

Reply:
ok